pyro/poutine/block_poutine.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, fn,
         1. msg["name"] in hide
         2. msg["type"] in hide_types
         3. msg["name"] not in expose and msg["type"] not in expose_types
-        4. hide_all == True
+        4. hide_all == True and hide, hide_types, and expose_types are all None
```

Comment:
I have to think really hard to guess what combinations of arguments are valid. Do you think it would be reasonable to eliminate some args? E.g. `hide_all=True` should be equivalent to `expose=[]`, and `expose_all=True` should be equivalent to `hide=[]`. Maybe `hide` would be better named `hide_only`?

Alternatively, what do you think of adding a stricter sanity checking block at the the start of the function, e.g.
```py
if hide_all and expose_all:
    raise ValueError('Cannot set both hide_all=True and expose_all=True")
if hide_all:
    if hide is not None:
        raise ValueError("Cannot set both hide_all and hide")
    if hide_types is not None:
        raise ValueError("Cannot set both hide_all and hide_types")
if expose_all:
    if expose is not None:
        raise ValueError("Cannot set both hide_all and hide")
    if expose_types is not None:
        raise ValueError("Cannot set both hide_all and hide_types")
```
This would at least make it clearer to developers which arg combos are valid.